filetests/isa/x86/simd-arithmetic-legalize.clif

```diff
@@ -39,13 +39,17 @@ function %fneg_legalized() {
 ebb0:
     v0 = vconst.f32x4 [0x1.0 0x2.0 0x3.0 0x4.0]
     v1 = fneg v0
-    ; check: v4 = vconst.f32x4 0x00
-    ; nextln: v1 = fsub v4, v0
+    ; check: v4 = vconst.i32x4 0xffffffffffffffffffffffffffffffff
+    ; nextln: v5 = ishl_imm v4, 31
+    ; nextln: v6 = raw_bitcast.f32x4 v5
+    ; nextln: v1 = bxor v0, v6
 
     v2 = vconst.f64x2 [0x1.0 0x2.0]
     v3 = fneg v2
-    ; check: v5 = vconst.f64x2 0x00
-    ; nextln: v3 = fsub v5, v2
+    ; check: v7 = vconst.i64x2 0xffffffffffffffffffffffffffffffff
+    ; nextln: v8 = ishl_imm v7, 63
+    ; nextln: v9 = raw_bitcast.f64x2 v8
+    ; nextln: v3 = bxor v2, v9
 
     return
```

Comment:
Can you add tests for -0, 0, maybe -Infinity, +Infinity, and one NaN value, please?

Reply:
Sure, but the most complete testing (which I believe covers those cases) is already present in the spec tests.

Reply:
I added a `run` test for `[0x0.0 -0x0.0 -Inf Inf]` but I can't use `NaN` in there yet because the comparison requires `ueq` (unordered or equal) which is not yet implemented.